lib/iris/analysis/trajectory.py

```diff
@@ -228,22 +240,29 @@ def interpolate(cube, sample_points, method=None):
     for coord, values in sample_points:
         if coord.ndim > 1:
             if method == "linear":
-                raise iris.exceptions.CoordinateMultiDimError("Cannot currently perform linear interpolation for multi-dimensional coordinates.")
+                raise iris.exceptions.CoordinateMultiDimError(
+                        "Cannot currently perform linear interpolation for "
+                        "multi-dimensional coordinates.")
             method = "nearest"
             break
 
     # Use a cache with _nearest_neighbour_indices_ndcoords()
     cache = {}
 
+    # Cache the linear interpolator
+    scheme = iris.analysis.Linear()
```

Comment:
This works fine, but it's not as quick (more like a 45% improvement), and the test still fails.